src/main/java/edith/Parser.java

```diff
@@ -0,0 +1,149 @@
+     */
+    public static String getTaskDetails(String userInput, String taskType) throws MissingTaskNameException {
+        try {
+            List<String> userInputs = Arrays.asList(userInput.split(taskType + " "));
+            return userInputs.get(1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new MissingTaskNameException();
+        }
+    }
+
+    /**
+     * Returns task name
+     * @param taskDetails Details of task in a string including task name and task deadline/duration if applicable.
+     * @param taskType Type of task (Todo, Deadline or Event).
+     * @return Task name in a string.
+     */
+    public static String getTaskName(String taskDetails, String taskType) {
+        List<String> userInputs;
+        if (Objects.equals(taskType, "deadline")) {
+            userInputs = Arrays.asList(taskDetails.split(" /by "));
+        } else {
+            userInputs = Arrays.asList(taskDetails.split(" /from "));
+        }
+        return userInputs.get(0);
+    }
+
+    /**
+     * Returns task deadline (only applicable to tasks of class DeadlineTask).
+     * @param taskDetails Details of task in a string including task name and task deadline.
+     * @return Task deadline in a string.
+     * @throws MissingDeadlineException When task deadline is not specified.
+     */
+    public static String getTaskDeadline(String taskDetails) throws MissingDeadlineException {
+        try {
+            List<String> userInputs = Arrays.asList(taskDetails.split(" /by "));
+            return userInputs.get(1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new MissingDeadlineException();
+        }
+    }
+
+    /**
+     * Returns task deadline (only applicable to tasks of class EventTask).
+     * @param taskDetails Details of task in a string including task name and task duration.
+     * @return Task duration in a string.
+     * @throws MissingEventDurationException When task duration is not specified.
+     */
+    public static String getTaskDuration(String taskDetails) throws MissingEventDurationException {
+        try {
+            List<String> userInputs = Arrays.asList(taskDetails.split(" /from "));
+            return userInputs.get(1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new MissingEventDurationException();
+        }
+    }
+
+    /**
+     * Returns task start date and/or time (only applicable to tasks of class EventTask).
+     * @param taskDuration Duration of task in a string including start and end date/time.
+     * @return Task start date and/or time in a string.
+     * @throws MissingEventDurationException When task duration is not specified.
+     */
+    public static String getTaskStart(String taskDuration) throws MissingEventDurationException {
+        try {
+            List<String> userInputs = Arrays.asList(taskDuration.split(" /to "));
+            return userInputs.get(0);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
There are methods in your code that are almost identical but with minor differences. For example, getTaskDeadline, getTaskDuration, getTaskStart, and getTaskEnd all have similar try-catch blocks. You could create a helper method that extracts parts of a string based on a delimiter, reducing duplication